full_storage_utilization_test.py

```diff
@@ -0,0 +1,149 @@
+import time
+from sdcm.tester import ClusterTester
+from sdcm.cluster import MAX_TIME_WAIT_FOR_NEW_NODE_UP
+
+class FullStorageUtilizationTest(ClusterTester):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def prepare_dataset_layout(self, dataset_size, row_size=10240):
+        n = dataset_size * 1024 * 1024 * 1024 // row_size
+        seq_end = n * 100
+
+        return f'cassandra-stress write cl=ONE n={n} -mode cql3 native -rate threads=10 -pop dist="uniform(1..{seq_end})" ' \
+               f'-col "size=FIXED({row_size}) n=FIXED(1)" -schema "replication(strategy=NetworkTopologyStrategy,replication_factor=3)"'
+
+    def setUp(self):
+        super().setUp()
+        self.start_time = time.time()
+
+    def test_storage_utilization(self):
+        """
+        3 nodes cluster, RF=3.
+        Write data until 85% disk usage is reached.
+        Sleep for 90 minutes.
+        Add 4th node.
+        """
+        self.log.info("Running df command on all nodes:")
+        self.get_df_output()
+        self.run_stress(85)
+        self.run_stress(93)
+
+        self.log.info("Adding a new node")
+        self.add_new_node()
```

Comment:
One major comment here. Adding a node, will always succeed because you no longer write data. Your implementation of `run_stress_until_target` to prepare a cluster is fine. Once this is achieved, you want to rather react based on the storage utilization.

I think, this part of adding a node shall be written differently. Namely, you want to monitor the disk usage while writing bunch of data and if a certain threshold is reached, you extend a cluster by adding a new node.

Reply:
Updated the PR, now after reaching 90% storage limit, script starts writing 10GB data over 30mins and add node is started. So there will be on-going write operation during topology change.